src/ranged.cpp

```diff
@@ -161,19 +161,58 @@ bool player::handle_gun_damage( item &it, int shots_fired )
 {
     if( !it.is_gun() ) {
         debugmsg( "Tried to handle_gun_damage of a non-gun %s", it.tname() );
+        shots_fired = shots_fired + 0; // left this here in case someone needs shots_fired in the future.
         return false;
     }
 
     const auto &curammo_effects = it.ammo_effects();
     const cata::optional<islot_gun> &firing = it.type->gun;
+    const std::string thistimefired = to_string_time_of_day(calendar::turn);
```

Comment:
This really isn't necessary (and might behave incorrectly in some edge cases). Better just declare a boolean variable outside of this function, specifying whether a gun is fired for the first time or not during the burst, then call it with that variable.

Reply:
Cycling doesn't just affect auto fire, it also affects semiauto, which is really just auto but with restrictions making you pull the trigger each time. In both cases it's still the gas generated by the fired round causing the weapon to cycle, and that's where the jamming occurs.